### PR TITLE
[graphql-js] Add individual validation rules and "getDirectiveValues"

### PR DESCRIPTION
**Pull request by IvanGoncharov**
Please fill in this template.
 
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/graphql/graphql-js/commit/a1f63083fb1abb7cb3d57754ae41a54c8ff4b382 and https://github.com/graphql/graphql-js/commit/2b94926ff62430f3679f2eb590dc4f8c6f594302
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by bowdenk7** [comment]
This is a sizable change. Added the listed authors (@TonyPythoneer, @calebmer, @intellix, @firede, @kepennar, @freiksenet ) to get some other eyes on it. 

Thanks!

File: types/graphql/index.d.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //                 Firede <https://github.com/firede>
```
Comment (minestarks) [inline types/graphql/index.d.ts]: Please update the version number for graphql. I think these apply to 0.11?

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by minestarks** [comment]
@TonyPythoneer, @calebmer, @intellix, @firede, @kepennar, @freiksenet any comments?

File: types/graphql/index.d.ts (hunk 1 above)
Comment (IvanGoncharov) [reply types/graphql/index.d.ts]: @minestarks Done ✅

**Review by minestarks — APPROVED** [review]

**Comment by DxCx** [comment]
hey @IvanGoncharov
since you already added the new API from graphql 0.11.0
can you please also fix `subscribe` and `createSourceEventStream ` signatures? (to return Promise of AsyncIterable instead)

**Comment by IvanGoncharov** [comment]
> since you already added the new API from graphql 0.11.0

@DxCx I didn't add anything that didn't exist before `0.10`. Just update number in response to the @minestarks comment.

> can you please also fix subscribe and createSourceEventStream signatures? (to return Promise of AsyncIterable instead)

This PR stuck as is. I don't want to make it even worse by adding more functionality that nobody will review. I just added things that I missing in my project and subscription isn't something that I'm using. 

In general, this `graphql` typings are very partial and missing a lot of things beyond subscriptions. This PR was a test to find out is possible to contribute this typings into the official package or should I maintain my personal extension.

**Comment by intellix** [comment]
This looks good to me :) Can add the other stuff in another PR I guess

**Comment by IvanGoncharov** [comment]
@intellix Thanks for taking time to review this PR.  I plan to create 

> This LGTM :)

@minestarks Can you please merge this PR?

**Comment by IvanGoncharov** [comment]
@minestarks Than you 🎉 🙌

**Comment by intellix** [comment]
Just linking this here since it was mentioned: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19518

**Pull request merged**